### ISSUE TITLE
PlayConfig.getOptional has been removed from PlayFramework in Master branch

### ISSUE THREAD

**Issue by SMcClelland** [playframework/play-mailer#64]
Hi,

The Play Mailer project is using 2.4.2 right now which should not have any issues, but I just wanted to note that in the Play Framework Master branch things have changed that would cause PlayConfig to break if run against Play Master instead of 2.4.2.
This pull request: https://github.com/playframework/playframework/pull/4588 has removed the getOptional calls that are used by MailerPlugin.scala (See: https://github.com/playframework/play-mailer/blob/35358de784cae2158d12579d0283b10a28dcb0a3/src/main/scala/play/api/libs/mailer/MailerPlugin.scala)

As such, at some point you will probably want to adjust the Play Mailer plugin accoringly.

Thanks,

-Sean

**Comment by Mogztter** [comment]
Good catch with a detailed explanation! Thanks @SMcClelland, would you like to send a pull request to fix that issue ?

**Comment by SMcClelland** [comment]
Hi Guillaume,

My apologies for not posting a pull request when I opened this up. I considered doing so but my lack of knowledge on the best way to fix it led me to not. I don't actually use the Play Mailer project myself and just happened across this when looking at an issue posted in the Play Framework discussion board. Additionally I am still in the learning stages when it comes to Scala so I am hoping there is a good fix that I just am ingorant to.

The changes I referenced in the master branch are not pushed to 2.4 so we could not just easily switch over to the get[Option[String]] approach. I suppose you could create a 2.4 branch and just fix it in Master, but I wasn't sure if that was something you would want to do.

I did check out the play-mailer codebase and attempted various ways of using config.get while wrapping it in Options, but in the end I couldn't get something that the Java and Scala code was happy with. (Tests were failing) This is most likely due to my inexperience with Scala and I hope you can think of an approach I have missed. If you have a suggestion on how to go about fixing this I would be glad to attempt it and post the pull request. 

Thanks,

-Sean

**Comment by godenji** [comment]
@SMcClelland thanks for opening this. I forked the plugin, changed `getOptional[T]` to `getOption[T]`, updated build.sbt version to 2.5.0-SNAPSHOT, then compiled + `+publishLocal`.

Updated my project's dep on the plugin to 3.0.2-SNAPSHOT; bingo, hellacious Monday begone.

I'll issue a PR if I have time, but above nips problem in the bud for any others building against master and depending on play mailer 3.0.1
